features/frameworks/unity/unity/unity_internals.h

```diff
@@ -785,6 +785,12 @@ extern const char UnityStrErr64[];
 #define UNITY_TEST_ASSERT_DOUBLE_IS_NOT_DETERMINATE(actual, line, message)                       UnityAssertDoubleSpecial((_UD)(actual), (message), (UNITY_LINE_TYPE)(line), UNITY_FLOAT_IS_NOT_DET)
 #endif
 
+/*-------------------------------------------------------
+ * Test Ignore conditions
+ *-------------------------------------------------------*/
+
+#define UNITY_TEST_IGNORE_UNLESS(condition, line, message)                                       if (condition) {} else {UNITY_TEST_IGNORE((UNITY_LINE_TYPE)(line), (message));}
+
```

Comment:
All macro params should be in ( ) ie if( (condition) ), but anyway could this not be rewritten as :
if ( !(condition)) {UNITY_TEST_IGNORE......}

Reply:
Agree about the parentheses, though I think I copied it from existing macros.
Regarding the condition: I preferred to keep this logic close to the `TEST_ASSERT` one, meaning that the condition inside the macro is the one reflecting the normal state.
So let's say I have test that asserts if I fail to allocate memory and I decide to ignore this instead.
So if this is initially
`TEST_ASSERT(memory_allocated)`
it will change to 
`TEST_IGNORE_UNLESS(memory_allocated)`

However, accepting the logic you suggest would mean that it should be:
`TEST_IGNORE(!memory_allocated)`
Which is counter intuitive IMO.